### ISSUE TITLE
Question: Explicitly manage a grain's lifecycle by actively choosing to remove it

### ISSUE THREAD

**Issue by sandervanteinde** [dotnet/orleans#8579]
Just a question here with regards to grain lifecycle management.

I have a use-case where I would like to explicitly free up resources by deleting/deactivating grains by their Id's

Is there an API available for this? Or am I overseeing some other method to implement similar behavior?

**Comment by ReubenBond** [comment]
Yes, you can tell individual grains to deactivate like so:
```C#
await myGrain.AsReference<IGrainManagementExtension>().DeactivateOnIdle()
```

If you wanted to deactivate all grains which have been idle for a certain period of time (eg, 5 mins), then you could instead do it this way:

```C#
await grainFactory.GetGrain<IManagementGrain>(0).ForceActivationCollection(TimeSpan.FromMinutes(5));
```

**Comment by sandervanteinde** [comment]
Hi @ReubenBond !

Thanks for the response, could you clarify what is considered `Idle`?

For instance, if I have a stream subscription active for that grain, is it then still idle if there are no events being processed?

**Comment by ReubenBond** [comment]
That's correct. A grain is idle when it is not processing a message (i.e, a grain call, which includes handling reminders and stream messages)

**Comment by sandervanteinde** [comment]
Awesome, thanks for letting me know. Would you like me to keep this issue open, since I see you've tagged it with `Needs: documentation` ?

**Comment by ReubenBond** [comment]
Yes, let's keep it open. Thanks. I think this is a good example of something which would go into a "how to" / "cookbook" style of documentation
